core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackendUtil.scala

```diff
@@ -0,0 +1,117 @@
+          case Array(container_path, "ro") =>
+            Some(vol.setContainerPath(container_path)
+                    .setMode(Volume.Mode.RO))
+          case Array(host_path, container_path) => 
+            Some(vol.setContainerPath(container_path)
+                    .setHostPath(host_path))
+          case Array(host_path, container_path, "rw") =>
+            Some(vol.setContainerPath(container_path)
+                    .setHostPath(host_path))
+          case Array(host_path, container_path, "ro") =>
+            Some(vol.setContainerPath(container_path)
+                    .setHostPath(host_path)
+                    .setMode(Volume.Mode.RO))
+          case spec => {
+            logWarning("parseVolumeSpec: unparseable: " + spec.mkString(":"))
+            None
+          }
+      }
+    }
+    .flatMap { _.map(_.build) }
+    .toList
+  }
+
+  /**
+   * Parse a portmap spec, similar to the form passed to 'docker run -p'
+   * the form accepted is host_port:container_port[:proto]
+   * Note:
+   * the docker form is [ip:]host_port:container_port, but the DockerInfo
+   * message has no field for 'ip', and instead has a 'protocol' field.
+   * Docker itself only appears to support TCP, so this alternative form
+   * anticipates the expansion of the docker form to allow for a protocol
+   * and leaves open the chance for mesos to begin to accept an 'ip' field
+   */
+  def parsePortMappingsSpec(portmaps: String): List[DockerInfo.PortMapping] = {
+    portmaps.split(",").map(_.split(":")).map { spec: Array[String] =>
+      val portmap: DockerInfo.PortMapping.Builder = DockerInfo.PortMapping
+        .newBuilder()
+        .setProtocol("tcp")
+      spec match {
+        case Array(host_port, container_port) =>
+          Some(portmap.setHostPort(host_port.toInt)
+                      .setContainerPort(container_port.toInt))
+        case Array(host_port, container_port, protocol) =>
+          Some(portmap.setHostPort(host_port.toInt)
+                      .setContainerPort(container_port.toInt)
+                      .setProtocol(protocol))
+        case spec => {
+          logWarning("parsePortMappingSpec: unparseable: " + spec.mkString(":"))
+          None
+        }
+      }
+    }
+    .flatMap { _.map(_.build) }
+    .toList
+  }
+
+  def withDockerInfo(
+      container: ContainerInfo.Builder,
+      image: String,
+      volumes: Option[List[Volume]] = None,
+      network: Option[ContainerInfo.DockerInfo.Network] = None,
+      portmaps: Option[List[ContainerInfo.DockerInfo.PortMapping]] = None):Unit = {
+
+    val docker = ContainerInfo.DockerInfo.newBuilder().setImage(image)
+    network.map(docker.setNetwork _)
+    portmaps.map(_.map(docker.addPortMappings _))
+
+    container.setType(ContainerInfo.Type.DOCKER)
+    container.setDocker(docker.build())
+    volumes.map(_.map(container.addVolumes _))
+
+    logInfo("withDockerInfo: using docker image: " + image)
```

Comment:
I think we should change the log level on saying which image we're using. I just launched a 1000 spark executor tasks with mesos, and it's showing up 1000 times that I'm using this image. And we only log it once, or change it to be debug?